daq/docker_test.py

```diff
@@ -100,15 +100,12 @@ def start(self, port, params, callback, finish_hook):
             raise e
         LOGGER.info("Target port %d test %s running", self.target_port, self.test_name)
 
-    def terminate(self, expected=True):
+    def terminate(self):
```

Comment:
Why is expected going away?

Reply:
The docker container status can be unknown which is why it’s best for docker_test to figure out if there is still a container to terminate. There’s an exception that’s thrown every time a host is shutdown caused by external factors like timeout. This fixes that problem.

Reply:
How is this case (ip flap timeout) fundamentally different than the module timeout case? At this level, it's a deterministic system so "can be unknown" is not a good sign. And "best" is subjective (and something I disagree with); certainly it's easier, but for a robust system that's often not the right thing. Before altering this, I'd want to know why/how this is fundamentally different than the other cases that happen, which don't put the system in an unknown state.

Reply:
The fundamental difference is that module timeout kills the test, and port flap timeout kills the host and all the remaining tests. When killing the host, which happens at the runner level, runner currently always assumes all test containers are terminated under 

> target_set_cancel -> target_host.terminate(... trigger=False)-> self.test_host.terminate(expected=False) -> if bool(self.docker_host) != expected: raise exception 

It is usually not the case during a port flap that a container is already terminated so that assumption causes an exception to be raised. To fix that issue, I considered moving that logic to host; however even at the host level, the container running status can be unknown. 
Host only knows the container termination after the docker_test callback; however, there is a lag between the container is terminated (stream hangup) and the callback is called. Depending on the order these events are processed, host may not know a running container is already terminated when it tries to terminate the container again.